[MRG]Changed default values of epoch parameters. Fix issue #2872

kaichogami opened the pull request:
Changed `event_id = None`, `tmin = -0.2`, `tmax=0.5`. I changed the value seeing the default in `mne/tests/test_epochs.py`. Reviews and suggestions about the default values please @agramfort @kingjr @teonlamont @jona-sassenhagen @Eric89GXL
closes #2872

jona-sassenhagen commented:
Good. Make a test that would have failed with the current code, and passes with yours.

dengemann commented:
Sorry, but why are we doing this? This is certainly an API change and I'm not sure what are the compelling reasons for it.

dengemann commented:
Ok forget what I said about the API change, here we're in Base epochs.

dengemann commented:
I'm still not convinced we need these defaults here, it will give us implicit behavior for for a step that should be rather controlled. I could see that for event id, the tmin and tmax are really parameters that should be chosen explicitly.

jasmainak commented:
@kaichogami when you are addressing an issue, make sure to use the keyword "closes #2872" in the PR description (not title). That way, github will automatically close the issue when the PR is merged.

jasmainak commented:
Also, please update `whats_new` API section.

jasmainak commented on mne/epochs.py:
```diff
@@ -144,7 +144,7 @@ class _BaseEpochs(ProjMixin, ContainsMixin, UpdateChannelsMixin,
     directly. See Epochs below for an explanation of the parameters.
     """
-    def __init__(self, info, data, events, event_id, tmin, tmax,
+    def __init__(self, info, data, events, event_id=None, tmin=-0.2, tmax=0.5,
```
make sure all events are selected when `event_id` is `None`

kaichogami replied:
I apologize as this might sound like a stupid question because of my inadequate knowledge, isn't it handled in `if event_id is None` part [here](https://github.com/mne-tools/mne-python/blob/master/mne/epochs.py#L163)?

teonlamont replied:
it's handled internally in the code, just not in the constructor

kaichogami replied:
Since subclasses, in their own constructor handle `tmax` and `tmin`, modifying to default here isn't necessary, although adds consistency.

kaichogami commented:
@jasmainak Yes, sorry for that. I will make sure to mention it in description next time.

teonlamont commented:
I'm in it for the default for `event_id` since there is a behavior already for `event_id=None`, i'm agnostic to `tmin`, `tmax`

kaichogami commented:
@teonlamont `event_id=None` is not possible unless we also make `tmin` and `tmax` have some default value. There is an ongoing discussion in the original issue #2872 for the best approach. 
@jasmainak I will update `whats_new` and update the changes here to add more verbosity as @dengemann mentioned in the issue tomorrow! 
Thanks for your comments! :)

kaichogami commented:
@jasmainak @teonlamont @dengemann @jona-sassenhagen I made some changes.
* Updated doc which mentions about the default value of the parameters. @jona-sassenhagen I saw your idea about using `warnings` in the issue, although I did not think it would be a good idea as mostly this will be used in tests. WDYT?
* Added tests, which compares `hash` of default parameters and same values provided explicitly. 

Please tell me your reviews and let me know what you think! 
Thank you!

agramfort commented on mne/tests/test_epochs.py:
```diff
@@ -1979,4 +1979,13 @@ def test_seeg():
+def test_default_values():
+    """Test default event_id, tmax tmin values are working correctly"""
+    raw, events = _get_data()[:2]
+    epoch_1 = Epochs(raw, events, preload=True)
```
please use 1 event only to speed it up

agramfort replied:
so events[:1]

agramfort commented:
travis is not happy

kaichogami commented:
@agramfort I made the changes you mentioned and travis looks happy. Please have a look when you are free.

agramfort commented:
Thx @kaichogami

teonlamont commented:
thanks @kaichogami!
